stdlib/email/src/main/java/org/ballerinalang/stdlib/email/client/SmtpClient.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.email.client;
+
+import org.ballerinalang.jvm.StringUtils;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.jvm.values.api.BValueCreator;
+import org.ballerinalang.stdlib.email.util.SmtpConstants;
+import org.ballerinalang.stdlib.email.util.SmtpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Properties;
+
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+
+/**
+ * Contains functionality of SMTP Client.
+ */
+public class SmtpClient {
+
+    private static final Logger log = LoggerFactory.getLogger(SmtpClient.class);
+
+    private SmtpClient() {
+        // Singleton class
+    }
+
+    /**
+     * Initializes the ObjectValue object with the SMTP Properties.
+     * @param clientEndpoint Represents the SMTP Client class
+     * @param config Properties required to configure the SMTP Session
+     * @return Object If an error occurs in the SMTP client, error
+     */
+    public static Object initClientEndpoint(ObjectValue clientEndpoint, MapValue<Object, Object> config) {
+        try {
+            log.debug("[SmtpClient][InitClient] Calling getProperties");
```

Comment:
Shouldn't we check `log.isDebugEnabled()` before debug logs?

Reply:
Changed all the debug locations as given.